backend/src/http/routes/users/auth.rs

```
@@ -43,26 +48,39 @@ pub(crate) async fn login(
     login_details: Result<Json<UserPayload<Login>>, JsonRejection>,
 ) -> Result<Json<UserPayload<User>>, Error> {
     let Json(UserPayload { user }) = login_details?;
-    // @Dzmitry, we of course should not be just dropping user's password,
-    // rather should verify it's not empty, hash it and compare to what is
-    // stored in our database
-    drop(user.password);
 
-    // @Dzmitry as if we found a user in db
-    let uid = Uuid::parse_str("25f75337-a5e3-44b1-97d7-6653ca23e9ee").unwrap();
+    // check email and password fields
+    user.validate()?;
 
-    // @Dzmitry and we issued a token for the newly created user
-    let jwt_string = issue_token(uid, &ctx.enc_key).unwrap();
+    let user_row = sqlx::query!(
+        r#"
+            SELECT user_id, username, email, bio, image, password_hash 
+            FROM users 
+            WHERE email = $1
+        "#,
+        &user.email
+    )
+    .fetch_one(&ctx.db)
+    .await
+    .map_err(|_e| Error::Unauthorized)?;
 
-    let paylaod = UserPayload {
+    let is_password_verified = verify_password(&user.password, &user_row.password_hash)?;
+
+    if !is_password_verified {
+        return Err(Error::Unauthorized);
+    }
+
+    let jwt_string = issue_token(user_row.user_id, &ctx.enc_key).unwrap();
+
+    let payload = UserPayload {
         user: User {
-            email: user.email,
+            email: user_row.email,
             token: jwt_string,
-            username: "rob.pike".into(),
-            bio: "Co-author Go programming language".into(),
-            image: Some("https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT_ySzB8CjQ85dLtvWXX8K1F4RlxbPEzjgfgKNTwneiPUCyfixt4edM8Nc&s".parse().expect("a valid URL")),
+            username: user_row.username,
+            bio: user_row.bio,
+            image: user_row.image.and_then(|img_str| Url::parse(&img_str).ok()),
```

I don't think we should fail silently if the image url is not a valid one. We can use [this approach](https://github.com/rustworthy/realworld-axum-react/blob/74cd82300a2de8ebfd70bd28c90c1a97ccb18bf2/backend/src/http/routes/articles/crud.rs#L103-L109) or at least log a warning that we failed to parse their image url